com.unity.multiplayer.mlapi/Runtime/Core/NetworkManager.cs

```diff
@@ -884,131 +898,86 @@ public void NetworkUpdate(NetworkUpdateStage updateStage)
 
 #if !UNITY_2020_2_OR_NEWER
                     NetworkProfiler.EndTick();
 #endif
 
 #if DEVELOPMENT_BUILD || UNITY_EDITOR
-                    s_ReceiveTick.End();
+                s_TransportPoll.End();
 #endif
-                }
             }
         }
 
+        // TODO Once we have a way to subscribe to NetworkUpdateLoop with order we can move this out of NetworkManager but for now this needs to be here because we need strict ordering.
         private void OnNetworkPreUpdate()
         {
-            if (IsListening)
-            {
-                if (((NetworkTime - m_LastEventTickTime >= (1f / NetworkConfig.EventTickrate))))
-                {
-#if DEVELOPMENT_BUILD || UNITY_EDITOR
-                    s_EventTick.Begin();
-#endif
-#if UNITY_EDITOR && !UNITY_2020_2_OR_NEWER
-                    NetworkProfiler.StartTick(TickType.Event);
-#endif
+            // Only update RTT here, server time is updated by time sync messages
+            NetworkTimeSystem.Advance(Time.deltaTime);
+            NetworkTickSystem.UpdateTick(NetworkTimeSystem.LocalTime, NetworkTimeSystem.ServerTime);
 
-                    if (NetworkConfig.EnableNetworkVariable)
-                    {
-                        // Do NetworkVariable updates
-                        BehaviourUpdater.NetworkBehaviourUpdate(this);
-                    }
-
-                    if (!IsServer && NetworkConfig.EnableMessageBuffering)
-                    {
-                        BufferManager.CleanBuffer();
-                    }
-
-                    if (IsServer)
-                    {
-                        m_LastEventTickTime = NetworkTime;
-                    }
-#if UNITY_EDITOR && !UNITY_2020_2_OR_NEWER
-                    NetworkProfiler.EndTick();
-#endif
-
-#if DEVELOPMENT_BUILD || UNITY_EDITOR
-                    s_EventTick.End();
-#endif
-                }
-
-                if (IsServer && NetworkConfig.EnableTimeResync && NetworkTime - m_LastTimeSyncTime >= NetworkConfig.TimeResyncInterval)
-                {
-#if UNITY_EDITOR && !UNITY_2020_2_OR_NEWER
-                    NetworkProfiler.StartTick(TickType.Event);
-#endif
-                    SyncTime();
-                    m_LastTimeSyncTime = NetworkTime;
-#if UNITY_EDITOR && !UNITY_2020_2_OR_NEWER
-                    NetworkProfiler.EndTick();
-#endif
-                }
-
-                if (!Mathf.Approximately(m_NetworkTimeOffset, m_CurrentNetworkTimeOffset))
-                {
-                    // Smear network time adjustments by no more than 200ms per second.  This should help code deal with
-                    // changes more gracefully, since the network time will always flow forward at a reasonable pace.
-                    float maxDelta = Mathf.Max(0.001f, 0.2f * Time.unscaledDeltaTime);
-                    m_CurrentNetworkTimeOffset += Mathf.Clamp(m_NetworkTimeOffset - m_CurrentNetworkTimeOffset, -maxDelta, maxDelta);
-                }
+            if (IsServer == false)
```

Comment:
Is this the still we like ? e.g. as opposed to `if (!IsServer)` ?

Reply:
We don't have a guideline for this and I prefer `== false` because it makes it easier to quickly skim over a piece of code and understand the flow correctly. A single `!` can easily missed.